php/libraries/NDB_BVL_Instrument.class.inc

```diff
@@ -476,14 +476,23 @@ class NDB_BVL_Instrument extends NDB_Page
      * Attempts to validate the form (using the defined rules) and
      * saves the validated data into the database
      *
-     * @return false if failure to save data
+     * @return bool false if failure to save data
      * @access public
      */
     function save()
     {
         $scoreResult = '';
         if ($this->form->validate()) {
-            $this->form->process(array(&$this, '_saveValues'), true);
+            $success = $this->form->process(array(&$this, '_saveValues'), true);
+            if (!is_bool($success)) {
+                throw new Exception(
```

Comment:
This should be a more specific exception type (at a minimum, a LorisException so that you can tell the exception came from Loris..)